packages/cipherstash-proxy/src/postgresql/column_mapper.rs

```diff
@@ -0,0 +1,167 @@
+use crate::{
+    eql::Identifier,
+    error::{EncryptError, Error},
+    log::MAPPER,
+    postgresql::Column,
+    proxy::EncryptConfig,
+};
+use eql_mapper::{EqlTerm, TableColumn, TypeCheckedStatement};
+use postgres_types::Type;
+use std::sync::Arc;
+use tracing::{debug, warn};
+
+/// Service responsible for processing columns from type-checked SQL statements
+/// and mapping them to encryption configurations.
+#[derive(Clone)]
+pub struct ColumnMapper {
+    encrypt_config: Arc<EncryptConfig>,
+}
+
+impl ColumnMapper {
```

Comment:
Functions responsible for mapping a column to configuration lifted straight from frontend.

I expect this to evolve with further refactoring.